internal/process/defunct_processes.go

```diff
@@ -0,0 +1,89 @@
+package process
+
+import (
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// ProcessFS is the process file system.
+const ProcessFS = "/proc"
+
+// Stat represents status information of a process from /proc/[pid]/stat.
+type Stat struct {
+	// Comm is the command name (usually the executable filename).
+	Comm string
+
+	// State is the state of the process.
+	State string
+}
+
+// DefunctProcesses returns the number of zombie processes in the node.
+func DefunctProcesses() (defunctCount uint, retErr error) {
+	directories, err := os.Open(ProcessFS)
+	if err != nil {
+		logrus.Warn(err)
+		return 0, err
+	}
+	defer directories.Close()
+
+	names, err := directories.Readdirnames(-1)
+	if err != nil {
+		logrus.Warn(err)
+		return 0, err
+	}
+
+	for _, name := range names {
+		// Processes have numeric names. If the name cannot
+		// be parsed to an int, it is not a process name.
+		_, err := strconv.ParseInt(name, 10, 0)
+		if err != nil {
```

Comment:
nit: line can be combined with previous one

Reply:
@saschagrunert Like this: `_, err := strconv.ParseInt(name, 10, 0); if err != nil { continue }`

Reply:
Yes, with the `if` at the beginning of the line.